mocks/service_mock.go

```diff
@@ -39,7 +39,9 @@ func (m *ServiceMock) StartServer() error {
 }
 
 func (m *ServiceMock) ShutdownServer() {
-	m.server.Shutdown(context.TODO())
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
This also looks a bit hacky. At least here should be a comment describing why we create a context and then immediately cancel it. 

Did you manage to find out why we still have active connections to mock server? Was it caused by not closed body in runner.go?

Reply:
> Was it caused by not closed body in runner.go

Nope, it's rare edge case, I found it when investigating where connection may leak, but dont find

> This also looks a bit hacky

There are exists another way, we can add new method:
```
func (m *ServiceMock) ShutdownServerContext(ctx context.Context) error
```
and than reuse it in `ShutdownServer`, like in others libraries, which migrated to context

Reply:
I added comment here and made another PR with `ShutdownServerContext` method (https://github.com/lamoda/gonkey/pull/59), both of this solutions ok for me, choose what you like best